src/libraries/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -70,7 +70,9 @@ public bool UnsafeConnectionNtlmAuthentication
                 {
                     return;
                 }
-                lock ((DisconnectResults as ICollection).SyncRoot)
+
+                var disconnectResults = DisconnectResults;
```

Comment:
The local variable isn't helpful as it doesn't contain any common expression.

Reply:
Definition of DisconnectResults 
``` c#
private Dictionary<ulong, DisconnectAsyncResult> DisconnectResults =>
    LazyInitializer.EnsureInitialized(ref _disconnectResults, () => new Dictionary<ulong, DisconnectAsyncResult>());
```

So every access of DisconnectResults is a call to LazyInitializer.EnsureInitialized.
So using the local variable only one call is necessary.

Reply:
HttpListener isn't so optimized. I don't think reducing several volatile reads can result in real world difference. Let's focus on fixing only.